v-next/hardhat-node-test-reporter/src/error-formatting.ts

```diff
@@ -312,3 +336,17 @@ export function formatLocation(
     return locationPath;
   }
 }
+
+function isNodeLocation(location: string, strict: boolean = false): boolean {
```

Comment:
We should probably documment what strict means, or we'll forget it.

Reply:
I removed the notion of a strict check in the newest iteration. Instead, I added a new function [hasNodeContext](https://github.com/NomicFoundation/hardhat/pull/5714/commits/bfe40154dd252547b04e67bd3b47f47d30ecadf8#diff-561a47ff066e39f4e07ec5fdc8265f35643511c1776154bd916966b8f2add17aR365) which checks whether the context of a reference is for one of the node types. Currently, for the purpose of the function, `Array`, `Promise` and `SafePromise` are considered node types. We can explicitly expand this definition as we find new cases we want to support. I think this approach is safer and easier to understand.